src/jupyterhub_cost_monitoring/const_usage.py

```diff
@@ -2,17 +2,32 @@
 Constants used to query Prometheus for JupyterHub usage data.
 """
 
+# Resolution for Prometheus queries
+# For MEMORY_REQUESTS_PER_USER this needs to be a small value because pods can be created and deleted frequently.
+# And we want to capture the memory requests for each pod as they are created.
 TIME_RESOLUTION = "5m"
 
 MEMORY_REQUESTS_PER_USER = """
-    sum(
-        kube_pod_container_resource_requests{resource=\"memory\", namespace=~\".*\", pod=~\"jupyter-.*\"}  * on (namespace, pod)
-        group_left(annotation_hub_jupyter_org_username) group(
-            kube_pod_annotations{namespace=~\".*\", annotation_hub_jupyter_org_username=~\".*\"}
+    label_replace(
+        sum(
+            kube_pod_container_resource_requests{resource="memory", namespace=~".*", pod=~"jupyter-.*"} * on (namespace, pod)
+            group_left(annotation_hub_jupyter_org_username) group(
+                kube_pod_annotations{namespace=~".*", annotation_hub_jupyter_org_username=~".*"}
             ) by (pod, namespace, annotation_hub_jupyter_org_username)
-    ) by (annotation_hub_jupyter_org_username, namespace)
+        ) by (annotation_hub_jupyter_org_username, namespace),
+        "username", "$1", "annotation_hub_jupyter_org_username", "(.*)"
+    )
 """
 
+STORAGE_USAGE_PER_USER = """
+    label_replace(
+        sum(dirsize_total_size_bytes{namespace=~".*"}) by (namespace, directory),
+        "username", "$1", "directory", "(.*)"
```

Comment:
`annotation_hub_jupyter_org_username` uses the unescaped version of the username, e.g. `jwong@2i2c.org`.

`directory` uses the escaped version of the username, e.g. `jwong-402i2c-2eorg`.

We can use the https://github.com/jupyterhub/escapism library to consolidate the usernames (let's present the unescaped version to Grafana if possible)

Reply:
Created https://github.com/2i2c-org/jupyterhub-cost-monitoring/issues/19 to track this issue.